### PR TITLE
Allow fine-grained control over rgahp SSH BatchMode (SOFTWARE-4239)

### PR DESCRIPTION
**Pull request by brianhlin**
Based off of https://github.com/htcondor/htcondor/pull/130 and some cleanup of file overrides and patches by moving them into their own dir.

### CODE REVIEW

File: overrides/remote_gahp [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+# Exits with 0 on success, 1 if wrong arguments, < 0 if ssh-add failed, > 0 if ssh failed
+################################################################################
+
+#Defaults
+REMOTE_CMD=""
+REMOTE_HOSTNAME=""
+REMOTE_USER=`whoami`
+# Fix the home directory when spawned from a root process
+unset HOME
+export HOME=`echo ~`
```
Comment (matyasselmeci) [inline overrides/remote_gahp:27]: Does this work??

**Comment by matyasselmeci** [comment]
Can you point me to the original version of remote_gahp so I can check out your changes?

**Comment by brianhlin** [comment]
Yeah, you can see the diff in the PR that I linked to above (https://github.com/htcondor/htcondor/pull/130)

File: overrides/remote_gahp (hunk 1 above)
Comment (brianhlin) [reply overrides/remote_gahp:27]: Presumably yes because this is what's in upstream :shrug:

**Review by matyasselmeci — APPROVED** [review]

**Pull request merged**